aptos-move/move-examples/mint_nft/sources/minting.move

```diff
@@ -0,0 +1,250 @@
+            token::create_token_mutability_config(
+                &vector<bool>[ false, false, false, false, true ]
+            ),
+            vector<String>[string::utf8(b"given_to")],
+            vector<vector<u8>>[bcs::to_bytes(&signer::address_of(origin))],
+            vector<String>[string::utf8(b"address")],
+        );
+
+        let public_key = std::option::extract(&mut ed25519::new_validated_public_key_from_bytes(public_key_bytes));
+
+        move_to(origin, CollectionTokenMinter {
+            public_key,
+            signer_cap,
+            token_data_id,
+            expiration_timestamp,
+            minting_enabled: true,
+            token_minting_events: account::new_event_handle<TokenMintingEvent>(&resource),
+        });
+    }
+
+    /// Set if minting is enabled for this collection token minter
+    public entry fun set_minting_enabled(minter: &signer, minting_enabled: bool) acquires CollectionTokenMinter {
+        let minter_address = signer::address_of(minter);
+        assert!(minter_address == @mint_nft, error::permission_denied(ENOT_AUTHORIZED));
+        let collection_token_minter = borrow_global_mut<CollectionTokenMinter>(minter_address);
+        collection_token_minter.minting_enabled = minting_enabled;
+    }
+
+    /// Set the expiration timestamp of this collection token minter
+    public entry fun set_timestamp(minter: &signer, expiration_timestamp: u64) acquires CollectionTokenMinter {
+        let minter_address = signer::address_of(minter);
+        assert!(minter_address == @mint_nft, error::permission_denied(ENOT_AUTHORIZED));
+        let collection_token_minter = borrow_global_mut<CollectionTokenMinter>(minter_address);
+        collection_token_minter.expiration_timestamp = expiration_timestamp;
+    }
+
+    /// Mint an NFT to the receiver.
+    /// `mint_proof_signature` is the `MintProofChallenge` signed by the receiver's private key
+    /// `public_key_bytes` is the public key of the receiver
+    /// `account_scheme` is the account scheme of the receiver (should be 0/ed25519 or 1/multi_ed25519)
+    public entry fun mint_NFT(receiver: &signer, mint_proof_signature: vector<u8>, public_key_bytes: vector<u8>, account_scheme: u8) acquires CollectionTokenMinter {
+        let receiver_addr = signer::address_of(receiver);
+
+        // get the collection minter and check if the collection minting is disabled or expired
+        let collection_token_minter = borrow_global_mut<CollectionTokenMinter>(@mint_nft);
+
+        assert!(timestamp::now_seconds() < collection_token_minter.expiration_timestamp, error::permission_denied(ECOLLECTION_EXPIRED));
+        assert!(collection_token_minter.minting_enabled, error::permission_denied(EMINTING_DISABLED));
+
+        // verify that the receiver wants to claim this NFT
+        verify_proof_of_knowledge(receiver_addr, mint_proof_signature, collection_token_minter.token_data_id, public_key_bytes, account_scheme);
```

Comment:
I don't think we want a multisigner for this. It should just be ed25119.

Reply:
I was just thinking that we can add multisig as an example to show how to do a multisig proof - we're just passing in one additional parameter (the `account_scheme`)